lib/puppet/parser/functions/is_domain_name.rb

```diff
@@ -32,6 +32,7 @@ module Puppet::Parser::Functions
     # Check the whole domain
     return false if domain.empty?
     return false if domain.length > domain_max_length
+    return false if /^\d+\.\d+\.\d+\.\d+$/.match(domain)
```

Comment:
Would it be appropriate to use `Resolv::IPv4::Regex` and `Resolv::IPv6::Regex` here to match both 4 and 6 ip addresses without having to write our own regexes?

Thanks.